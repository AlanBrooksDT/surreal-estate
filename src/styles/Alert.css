.alert-success {
    background-color: seagreen;
    padding: 10px;
    margin: 10px;
}

.alert-error {
    background-color: red;
    padding: 10px;
    margin: 10px;
}