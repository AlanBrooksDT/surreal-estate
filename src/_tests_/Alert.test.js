import React from 'react';
import { render } from '@testing-library/react';
import Alert from '../components/Alert';


describe("Alert", () => {

    it('confirm error being flagged if post does not complete', () => {
        const component = render(<Alert message="Error!" />);
        const alertMessageNode = component.getByText("Error!")

        expect(alertMessageNode.textContent).toBe('Error!');
    });

    it("displays message if a success", () => {
        const component = render(<Alert message="Success!!!!" success />);
        const alertMessageNode = component.getByText("Success!!!!")
      
        expect(alertMessageNode.textContent).toBe("Success!!!!");
    });

    it("no error if property added successfully", () => {
        const { asFragment } = render(<Alert message="" />);
        const alert = asFragment();

        expect(alert).toMatchSnapshot();
    });

    it("display error message if property not added successfully", () => {
        const { getByText, asFragment } = render(<Alert message="Error!" />);

        expect(asFragment()).toMatchSnapshot();
        expect(getByText(/Error/).textContent).toBe("Error!");
    });

    it("display success message if property is added successfully", () => {
        const { getByText, asFragment } = render(<Alert message="Success!!!!" />);

        expect(asFragment()).toMatchSnapshot();
        expect(getByText(/Success/).textContent).toBe("Success!!!!");
    });
});