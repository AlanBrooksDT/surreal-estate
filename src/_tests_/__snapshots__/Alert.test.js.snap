// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert display error message if property not added successfully 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`Alert display success message if property is added successfully 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    Success!!!!
  </div>
</DocumentFragment>
`;

exports[`Alert no error if property added successfully 1`] = `<DocumentFragment />`;
